### PR TITLE
Add basic DocC archive.

### PR DESCRIPTION
**Pull request by jeroenleenarts**
### 🔗 Issue Links

_Provide all Jira tickets and/or Github issues related to this PR, if applicable._

### 🎯 Goal

_Describe why we are making this change._

### 📝 Summary

_Provide bullet points with the most important changes in the codebase._

### 🛠 Implementation

_Provide a detailed description of the implementation and explain your decisions if you find them relevant._

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

| Before | After |
| ------ | ----- |
|  image   |  image  |

### 🧪 Manual Testing Notes

_Explain how this change can be tested manually, if applicable._

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

**Comment by jeroenleenarts** [comment]
This could use some additional work. @martinmitrevski and @DaemonLoki this is the type of low hanging fruit we should harvest.

Adding a docc archive should add links here: https://swiftpackageindex.com/GetStream/stream-chat-swift/4.21.1/documentation/streamchat

File: Documentation.docc/Documentation.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# ``StreamChat``
+
+SummaryThis is the official iOS SDK for [Stream Chat](https://getstream.io/chat/sdk/ios/), a service for building chat and messaging applications. This library includes both a low-level SDK and a set of reusable UI components.
```
Comment (nuno-vieira) [inline Documentation.docc/Documentation.md:3]: Typo here:
```suggestion
This is the official iOS SDK for [Stream Chat](https://getstream.io/chat/sdk/ios/), a service for building chat and messaging applications. This library includes both a low-level SDK and a set of reusable UI components.
```

**Review by nuno-vieira — COMMENTED** [review]
Small typo, other than that, LGTM

**Review by polqf — APPROVED** [review]

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Comment by DaemonLoki** [comment]
We could use the aside feature to add some links to our tutorials maybe?
Something like:

> Tip: Not sure where to get started: check out our tutorial for [UIKit](https://getstream.io/tutorials/ios-chat/) or [SwiftUI](https://getstream.io/tutorials/swiftui-chat/)

[Reference how it's rendering in DocC](https://developer.apple.com/documentation/xcode/formatting-your-documentation-content)

**Pull request merged**